extension/data.ts

```diff
@@ -296,93 +287,308 @@ class RcxDict {
           }
         }
       }
-    } while (++i < r.length);
-
+    }
     return r;
   }
 
-  // katakana -> hiragana conversion tables
-  ch: number[] = [
-    0x3092, 0x3041, 0x3043, 0x3045, 0x3047, 0x3049, 0x3083, 0x3085, 0x3087,
-    0x3063, 0x30fc, 0x3042, 0x3044, 0x3046, 0x3048, 0x304a, 0x304b, 0x304d,
-    0x304f, 0x3051, 0x3053, 0x3055, 0x3057, 0x3059, 0x305b, 0x305d, 0x305f,
-    0x3061, 0x3064, 0x3066, 0x3068, 0x306a, 0x306b, 0x306c, 0x306d, 0x306e,
-    0x306f, 0x3072, 0x3075, 0x3078, 0x307b, 0x307e, 0x307f, 0x3080, 0x3081,
-    0x3082, 0x3084, 0x3086, 0x3088, 0x3089, 0x308a, 0x308b, 0x308c, 0x308d,
-    0x308f, 0x3093,
-  ];
+  hiraganaLookup: Record<string, string> = {
+    ぁ: 'ぁ',
+    あ: 'あ',
+    ぃ: 'ぃ',
+    い: 'い',
+    ぅ: 'ぅ',
+    う: 'う',
+    ぇ: 'ぇ',
+    え: 'え',
+    ぉ: 'ぉ',
+    お: 'お',
+    か: 'か',
+    が: 'が',
```

Comment:
I think the will match because in the case of half-width characters (the ones where this matters), we check the trailing character to see if it is a voice/semi-voiced mark. if so, we replace the previously added string with a new lookup combining the 2 characters together.

like this:
```
const key = isSemiVoiced
        ? char + nextChar
        : isVoiced
        ? char + nextChar
        : char;
      const hiragana = kanaToHiraganaNormalizationMap[key];

```